### ISSUE TITLE
[vcpkg] x-history command fails for boost package

### ISSUE THREAD

**Issue by Chronial** [microsoft/vcpkg#11422]
**To Reproduce**
1. ./vcpkg x-history boost
2. See error

**Failure output** 
```
Found 2 sets of Version: .*
 but expected at most 1, in block:
Source: boost
<<<<<<< HEAD
Version: 1.65-2
=======
Version: 1.65.1
>>>>>>> master
Description: Peer-reviewed portable C++ source libraries
Build-Depends: zlib, bzip2, icu [windows], python3 [windows]
```

**Additional context**
Looks like somebody checked in conflict markers at some point in the past and now `x-history` can't parse the Controlfile of that commit

**Comment by timautry** [comment]
- I get the same results. It only seems to happen on boost.
- Just a guess but maybe because boost is just a placeholder added when any of the individual boost ports are installed?
- When tried against any of the individual boost ports it works as it should?

`SYSTEM@ANDROMEDA 05/18/2020 13:01:04 D:\Git\vs2019.prod`
`->vcpkg x-history boost`
`Found 2 sets of Version: .*`
` but expected at most 1, in block:`
`Source: boost`
`<<<<<<< HEAD`
`Version: 1.65-2`
`=======`
`Version: 1.65.1`
`>>>>>>> master`
`Description: Peer-reviewed portable C++ source libraries`
`Build-Depends: zlib, bzip2, icu [windows], python3 [windows]`

`SYSTEM@ANDROMEDA 05/18/2020 13:01:14 D:\Git\vs2019.prod`
`->vcpkg x-history boost-regex`
`             version          date    vcpkg commit`
`    2020-05-13    24e8be5fcfddd0da98b58a63cd918849787fb06f`
`    2020-01-16    c50852aba0aa8df22e17133352edb432a9fc1e56`
`    2019-09-27    9433136b22b5e443d550ca7665a562637c8a86b9`
`    2019-06-15    18b029a5e3997fa4fdc7d3d06d56568a1d6f74ad`
`    2019-01-30    0e759709ddc7dbad0b881f9f068f1b04b2092773`
`    2018-08-20    eae02c95b33ec0272a232395820379e0a8b2393d`
`    2018-04-19    751015f6a44d6f89ca71ac63d7c5cab83cf531c0`
`    2018-04-19    b831381cf569436177ada02366c4850a665b8465`

`SYSTEM@ANDROMEDA 05/18/2020 13:02:06 D:\Git\vs2019.prod`
`->vcpkg x-history boost-random`
`             version          date    vcpkg commit`
`    2020-05-13    24e8be5fcfddd0da98b58a63cd918849787fb06f`
`    2020-01-16    c50852aba0aa8df22e17133352edb432a9fc1e56`
`    2019-09-27    9433136b22b5e443d550ca7665a562637c8a86b9`
`    2019-06-15    18b029a5e3997fa4fdc7d3d06d56568a1d6f74ad`
`    2019-01-30    0e759709ddc7dbad0b881f9f068f1b04b2092773`
`    2018-08-20    eae02c95b33ec0272a232395820379e0a8b2393d`
`    2018-04-19    751015f6a44d6f89ca71ac63d7c5cab83cf531c0`
`    2018-04-19    b831381cf569436177ada02366c4850a665b8465`

`SYSTEM@ANDROMEDA 05/18/2020 13:02:18 D:\Git\vs2019.prod`
`->vcpkg x-history boost-mpi`
`             version          date    vcpkg commit`
`    2020-05-13    24e8be5fcfddd0da98b58a63cd918849787fb06f`
`    2020-01-16    c50852aba0aa8df22e17133352edb432a9fc1e56`
`    2019-09-27    9433136b22b5e443d550ca7665a562637c8a86b9`
`    2019-07-31    455223d009fc6fef0d9293e511d64e9e372abdb0`
`    2019-06-15    18b029a5e3997fa4fdc7d3d06d56568a1d6f74ad`
`    2019-01-30    0e759709ddc7dbad0b881f9f068f1b04b2092773`
`    2018-08-20    eae02c95b33ec0272a232395820379e0a8b2393d`
`    2018-04-19    751015f6a44d6f89ca71ac63d7c5cab83cf531c0`
`    2018-04-19    b831381cf569436177ada02366c4850a665b8465`
`    2018-01-17    78ac1d0bb641619094772e942673094b14935e7f`

**Comment by Chronial** [comment]
This is the commit that causes the issue: https://github.com/microsoft/vcpkg/commit/ee2831c548b54cdccf064663b3fffda5d6a4c6b8#diff-b382e486803ef61f2e6c2d8e4111a98f

x-history should probably just silently ignore commits that have an unparsable `CONTROL` file?

**Comment by vicroms** [comment]
This was fixed in #13777
